protocol/contracts/templegold/SpiceAuction.sol

```diff
@@ -0,0 +1,319 @@
+        daoExecutor = _daoExecutor;
+        emit DaoExecutorSet(_daoExecutor);
+    }
+
+    /**
+     * @notice Set config for an epoch. This enables dynamic and multiple auctions especially for vested scenarios
+     * @dev Must be set before epoch auction starts
+     * @param _config Config to set
+     */
+    function setAuctionConfig(SpiceAuctionConfig calldata _config) external onlyDAOExecutor {
+        /// @dev epoch Id is only updated when auction starts. 
+        /// @dev cannot set config for past or ongoing auction
+        uint256 currentEpochIdCache = _currentEpochId;
+        if (currentEpochIdCache > 0) {
+            EpochInfo storage info = epochs[currentEpochIdCache];
+            /// Cannot set config for ongoing auction
+            if (info.isActive()) { revert InvalidConfigOperation(); }
+        }
+        if (_config.duration < MINIMUM_AUCTION_PERIOD 
+            || _config.duration > MAXIMUM_AUCTION_DURATION
+            || _config.waitPeriod > MAXIMUM_AUCTION_WAIT_PERIOD) { revert CommonEventsAndErrors.InvalidParam(); }
+        /// @dev startCooldown can be zero
+        if (_config.waitPeriod == 0
+            || _config.minimumDistributedAuctionToken == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        if (_config.recipient == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+
+        currentEpochIdCache += 1;
+        auctionConfigs[currentEpochIdCache] = _config;
+        emit AuctionConfigSet(currentEpochIdCache, _config);
+    }
+
+    /// @notice Remove auction config set for last epoch
+    function removeAuctionConfig() external override onlyDAOExecutor {
+        /// only delete latest epoch if auction is not started
+        uint256 id = _currentEpochId;
+        
+        EpochInfo storage info = epochs[id];
+        // _currentEpochId = 0
+        if (info.startTime == 0) { revert InvalidConfigOperation(); }
+        // Cannot reset an ongoing auction
+        if (info.isActive()) { revert InvalidConfigOperation(); }
+        /// @dev could be that `auctionStart` is triggered but there's cooldown, which is not reached (so can delete epochInfo for _currentEpochId)
+        // or `auctionStart` is not triggered but `auctionConfig` is set (where _currentEpochId is not updated yet)
+        SpiceAuctionConfig storage config = auctionConfigs[id];
+        bool configSetButAuctionStartNotCalled = auctionConfigs[id+1].duration > 0;
+        if (!configSetButAuctionStartNotCalled) {
+            /// auction was started but cooldown has not passed yet
+            delete auctionConfigs[id];
+            delete epochs[id];
+            _currentEpochId = id - 1;
+            emit AuctionConfigRemoved(id, id);
+        } else {
+            // `auctionStart` is not triggered but `auctionConfig` is set
+            id += 1;
+            config = auctionConfigs[id];
+            delete auctionConfigs[id];
+            emit AuctionConfigRemoved(id, 0);
+        }
```

Comment:
Let's imagine:
1. _currentEpochId = 100
2. auctionConfig for epoch#101 has been set but auctionStart is not triggered yet
3. If we call this `removeAuctionConfig` then, it will remove the auctionConfig[101]
4. If we call `removeAuctionConfig` again(I know this can be called by DAOExecutor but there would be some unexpected happening), then it will remove the auctionConfig[100] which already completed and furthermore `_currentEpochId` becomes 99
5. And DAOExecutor can call `removeAuctionConfig` functions till `_currentEpochId` becomes 0 which we lost all the auction info.

I think it's better to add filter not to remove already completed auctions.

Reply:
Good catch!